src/parse.coffee

```diff
@@ -33,8 +33,7 @@ parse = (source, callback) ->
       err = new Error(err.message)
 
     if apiElements
-      apiElements = apiElements.toRefract()
-      apiElements.content = apiElements.content.concat(annotations)
+      apiElements.content.unshift(warning) if warning
```

Comment:
I made a simple comparison using following code:

```coffee
console.log(JSON.stringify(apiElements.toRefract(), null, 2))
```

### apiElements.content.unshift()

Produces tree which works and passes all tests.

```json
{
  "element": "parseResult",
  "meta": {},
  "attributes": {},
  "content": [
    {
      "element": "annotation",
      "meta": {
        "classes": [
          "warning"
        ]
      },
      "attributes": {},
      "content": "Could not recognize API description format. Falling back to API Blueprint by default."
    },
    {
      "element": "category",
      "meta": {
        "classes": [
          "api"
        ],
        "title": ""
      },
      "attributes": {},
      "content": []
    }
  ]
}
```

### apiElements.unshift()

Produces tree which does not work and fails some tests.

```json
{
  "element": "parseResult",
  "meta": {},
  "attributes": {},
  "content": [
    {
      "element": "object",
      "meta": {},
      "attributes": {},
      "content": [
        {
          "element": "member",
          "meta": {},
          "attributes": {},
          "content": {
            "key": {
              "element": "string",
              "meta": {},
              "attributes": {},
              "content": "content"
            },
            "value": {
              "element": "string",
              "meta": {},
              "attributes": {},
              "content": "Could not recognize API description format. Falling back to API Blueprint by default."
            }
          }
        },
        {
          "element": "member",
          "meta": {},
          "attributes": {},
          "content": {
            "key": {
              "element": "string",
              "meta": {},
              "attributes": {},
              "content": "_storedElement"
            },
            "value": {
              "element": "string",
              "meta": {},
              "attributes": {},
              "content": "annotation"
            }
          }
        },
        {
          "element": "member",
          "meta": {},
          "attributes": {},
          "content": {
            "key": {
              "element": "string",
              "meta": {},
              "attributes": {},
              "content": "_meta"
            },
            "value": {
              "element": "object",
              "meta": {},
              "attributes": {},
              "content": [
                {
                  "element": "member",
                  "meta": {},
                  "attributes": {},
                  "content": {
                    "key": {
                      "element": "string",
                      "meta": {},
                      "attributes": {},
                      "content": "content"
                    },
                    "value": {
                      "element": "array",
                      "meta": {},
                      "attributes": {},
                      "content": [
                        {
                          "element": "object",
                          "meta": {},
                          "attributes": {},
                          "content": [
                            {
                              "element": "member",
                              "meta": {},
                              "attributes": {},
                              "content": {
                                "key": {
                                  "element": "string",
                                  "meta": {},
                                  "attributes": {},
                                  "content": "content"
                                },
                                "value": {
                                  "element": "object",
                                  "meta": {},
                                  "attributes": {},
                                  "content": [
                                    {
                                      "element": "member",
                                      "meta": {},
                                      "attributes": {},
                                      "content": {
                                        "key": {
                                          "element": "string",
                                          "meta": {},
                                          "attributes": {},
                                          "content": "key"
                                        },
                                        "value": {
                                          "element": "object",
                                          "meta": {},
                                          "attributes": {},
                                          "content": [
                                            {
                                              "element": "member",
                                              "meta": {},
                                              "attributes": {},
                                              "content": {
                                                "key": {
                                                  "element": "string",
                                                  "meta": {},
                                                  "attributes": {},
                                                  "content": "content"
                                                },
                                                "value": {
                                                  "element": "string",
                                                  "meta": {},
                                                  "attributes": {},
                                                  "content": "classes"
                                                }
                                              }
                                            },
                                            {
                                              "element": "member",
                                              "meta": {},
                                              "attributes": {},
                                              "content": {
                                                "key": {
                                                  "element": "string",
                                                  "meta": {},
                                                  "attributes": {},
                                                  "content": "_storedElement"
                                                },
                                                "value": {
                                                  "element": "string",
                                                  "meta": {},
                                                  "attributes": {},
                                                  "content": "string"
                                                }
                                              }
                                            }
                                          ]
                                        }
                                      }
                                    },
                                    {
                                      "element": "member",
                                      "meta": {},
                                      "attributes": {},
                                      "content": {
                                        "key": {
                                          "element": "string",
                                          "meta": {},
                                          "attributes": {},
                                          "content": "value"
                                        },
                                        "value": {
                                          "element": "object",
                                          "meta": {},
                                          "attributes": {},
                                          "content": [
                                            {
                                              "element": "member",
                                              "meta": {},
                                              "attributes": {},
                                              "content": {
                                                "key": {
                                                  "element": "string",
                                                  "meta": {},
                                                  "attributes": {},
                                                  "content": "_meta"
                                                },
                                                "value": {
                                                  "element": "object",
                                                  "meta": {},
                                                  "attributes": {},
                                                  "content": [
                                                    {
                                                      "element": "member",
                                                      "meta": {},
                                                      "attributes": {},
                                                      "content": {
                                                        "key": {
                                                          "element": "string",
                                                          "meta": {},
                                                          "attributes": {},
                                                          "content": "content"
                                                        },
                                                        "value": {
                                                          "element": "array",
                                                          "meta": {},
                                                          "attributes": {},
                                                          "content": []
                                                        }
                                                      }
                                                    },
                                                    {
                                                      "element": "member",
                                                      "meta": {},
                                                      "attributes": {},
                                                      "content": {
                                                        "key": {
                                                          "element": "string",
                                                          "meta": {},
                                                          "attributes": {},
                                                          "content": "_storedElement"
                                                        },
                                                        "value": {
                                                          "element": "string",
                                                          "meta": {},
                                                          "attributes": {},
                                                          "content": "object"
                                                        }
                                                      }
                                                    }
                                                  ]
                                                }
                                              }
                                            },
                                            {
                                              "element": "member",
                                              "meta": {},
                                              "attributes": {},
                                              "content": {
                                                "key": {
                                                  "element": "string",
                                                  "meta": {},
                                                  "attributes": {},
                                                  "content": "_attributes"
                                                },
                                                "value": {
                                                  "element": "object",
                                                  "meta": {},
                                                  "attributes": {},
                                                  "content": [
                                                    {
                                                      "element": "member",
                                                      "meta": {},
                                                      "attributes": {},
                                                      "content": {
                                                        "key": {
                                                          "element": "string",
                                                          "meta": {},
                                                          "attributes": {},
                                                          "content": "content"
                                                        },
                                                        "value": {
                                                          "element": "array",
                                                          "meta": {},
                                                          "attributes": {},
                                                          "content": []
                                                        }
                                                      }
                                                    },
                                                    {
                                                      "element": "member",
                                                      "meta": {},
                                                      "attributes": {},
                                                      "content": {
                                                        "key": {
                                                          "element": "string",
                                                          "meta": {},
                                                          "attributes": {},
                                                          "content": "_storedElement"
                                                        },
                                                        "value": {
                                                          "element": "string",
                                                          "meta": {},
                                                          "attributes": {},
                                                          "content": "object"
                                                        }
                                                      }
                                                    }
                                                  ]
                                                }
                                              }
                                            },
                                            {
                                              "element": "member",
                                              "meta": {},
                                              "attributes": {},
                                              "content": {
                                                "key": {
                                                  "element": "string",
                                                  "meta": {},
                                                  "attributes": {},
                                                  "content": "content"
                                                },
                                                "value": {
                                                  "element": "array",
                                                  "meta": {},
                                                  "attributes": {},
                                                  "content": [
                                                    {
                                                      "element": "object",
                                                      "meta": {},
                                                      "attributes": {},
                                                      "content": [
                                                        {
                                                          "element": "member",
                                                          "meta": {},
                                                          "attributes": {},
                                                          "content": {
                                                            "key": {
                                                              "element": "string",
                                                              "meta": {},
                                                              "attributes": {},
                                                              "content": "content"
                                                            },
                                                            "value": {
                                                              "element": "string",
                                                              "meta": {},
                                                              "attributes": {},
                                                              "content": "warning"
                                                            }
                                                          }
                                                        },
                                                        {
                                                          "element": "member",
                                                          "meta": {},
                                                          "attributes": {},
                                                          "content": {
                                                            "key": {
                                                              "element": "string",
                                                              "meta": {},
                                                              "attributes": {},
                                                              "content": "_storedElement"
                                                            },
                                                            "value": {
                                                              "element": "string",
                                                              "meta": {},
                                                              "attributes": {},
                                                              "content": "string"
                                                            }
                                                          }
                                                        }
                                                      ]
                                                    }
                                                  ]
                                                }
                                              }
                                            },
                                            {
                                              "element": "member",
                                              "meta": {},
                                              "attributes": {},
                                              "content": {
                                                "key": {
                                                  "element": "string",
                                                  "meta": {},
                                                  "attributes": {},
                                                  "content": "_storedElement"
                                                },
                                                "value": {
                                                  "element": "string",
                                                  "meta": {},
                                                  "attributes": {},
                                                  "content": "array"
                                                }
                                              }
                                            }
                                          ]
                                        }
                                      }
                                    }
                                  ]
                                }
                              }
                            },
                            {
                              "element": "member",
                              "meta": {},
                              "attributes": {},
                              "content": {
                                "key": {
                                  "element": "string",
                                  "meta": {},
                                  "attributes": {},
                                  "content": "_storedElement"
                                },
                                "value": {
                                  "element": "string",
                                  "meta": {},
                                  "attributes": {},
                                  "content": "member"
                                }
                              }
                            }
                          ]
                        }
                      ]
                    }
                  }
                },
                {
                  "element": "member",
                  "meta": {},
                  "attributes": {},
                  "content": {
                    "key": {
                      "element": "string",
                      "meta": {},
                      "attributes": {},
                      "content": "_storedElement"
                    },
                    "value": {
                      "element": "string",
                      "meta": {},
                      "attributes": {},
                      "content": "object"
                    }
                  }
                }
              ]
            }
          }
        }
      ]
    },
    {
      "element": "category",
      "meta": {
        "classes": [
          "api"
        ],
        "title": ""
      },
      "attributes": {},
      "content": []
    }
  ]
}
```

------

Just for the record, I used the `Dredd Transactions > When given no input > API Blueprint` test in `test/integration/dredd-transactions-test.coffee` to generate these outputs.